app/views/merge_requests/details.html.erb

```diff
@@ -0,0 +1,64 @@
+<% content_for :before_content do %>
+  <% title = "Merge details: #{@merge_request.absorbing_organisation_name}" %>
+  <% content_for :title, title %>
+  <%= govuk_back_link href: "#{organisations_path}#merge-requests" %>
+<% end %>
+<% unless @merge_request.signed_dsa || @merge_request.absorbing_organisation_id.blank? %>
+  <div class="govuk-notification-banner" role="region" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
+    <div class="govuk-notification-banner__header">
+      <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
+        Important
+      </h2>
+    </div>
+    <div class="govuk-notification-banner__content">
+      <strong>The absorbing organisation must accept the Data Sharing Agreement before merging.</strong>
+      <br>
+      <% if @merge_request.dpo_user %>
+        Contact the Data Protection Officer: <%= link_to @merge_request.dpo_user.name, user_path(@merge_request.dpo_user.id) %>
+      <% else %>
+        <%= @merge_request.absorbing_organisation_name %> does not have a Data Protection Officer. You can assign one on the <%= link_to 'users page', "#{organisation_path(@merge_request.absorbing_organisation_id)}/users" %>.
+      <% end %>
+    </div>
+  </div>
+<% end %>
```

Comment:
I added the boolean `signed_dsa` to the merge request model and use this to check if the DSA has been signed. This notification is also not shown in the edge case that the absorbing organisation has not yet been set. Determining who the DPO user is relies on there being an organisation first.